facade/src/main/java/org/jboss/pnc/facade/providers/BuildProviderImpl.java

```diff
@@ -227,7 +227,13 @@ public String getBuildLog(String id) {
     @Override
     public SSHCredentials getSshCredentials(String id) {
         BuildRecord buildRecord = getBuildRecord(id);
-        if(!buildRecord.getUser().equals(userService.currentUser())){
+        User user = null;
+        try {
+            user = userService.currentUser();
+        } catch(IllegalStateException e) {
```

Comment:
I would let it throw.

Reply:
If I let it throw, the Rest will return 500SystemError and not 403Forbidden as it should be.

Reply:
It should not throw 403 Forbidden, because the user is authenticated.

Reply:
It should throw 403 Forbidden because the user is not the one from BR and not because he is not authenticated. In reality, this should never happen because if a user built something, his entity is in the BuildRecord.

Reply:
still makes problems in integration-tests though

Reply:
You are right. And I mistook 401 Unautorized and 403 Forbidden again.